staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,614 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/polymorphichelpers"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+const (
+	logsNodesUsageStr         = "node-logs ([--role ROLE] | [--label LABEL] | [NODE...])"
+	logsNodesWriterBufferSize = 16384
+	logsNodesReaderBufferSize = 4096
+)
+
+var (
+	logsNodesLong = templates.LongDesc(i18n.T(`
+		Experimental: Display and filter node logs when the kubelet NodeLogs feature
+		flag is set.
+
+		This command retrieves logs for the node. The default mode is to query the
+		systemd journal or WinEvent entries on supported operating systems,
+		which allows searching, time based filtering, and service based filtering.
+		You may also use the --path argument to see a list of log files available
+		under /var/logs/ and view those contents directly.
+
+		Node logs may contain sensitive output and so are limited to privileged node
+		administrators. The system:node-admins role grants this permission by default.`))
+
+	logsNodesExample = templates.Examples(i18n.T(`
+		# Show kubelet and crio journald logs from all masters
+		kubectl alpha node-logs --role master -u kubelet -u crio
+
+		# Show kubelet log file (/var/log/kubelet/kubelet.log) from all Windows worker nodes
+		kubectl alpha node-logs --label kubernetes.io/os=windows -u kubelet
+
+		# Display docker WinEvent logs from a specific Windows worker node
+		kubectl alpha node-logs <node-name> --service docker
+
+		# Show crio journald logs from a specific Linux node
+		kubectl alpha node-logs <node-name> --service crio`))
```

Comment:
This should be modified because --service and --path were collapsed into --query

Reply:
Thanks for catching that. I have fixed it in [49cfc73](https://github.com/kubernetes/kubernetes/pull/96120/commits/49cfc734b27bb1006966aea00e30264f49b4a54c)